Refactor documentation and add project overview sections (#19)
eProsima/DDS-Recorder: docs/rst/recording/getting_started/project_overview.rst

This is achieved through the internal implementation of a zero-copy communication mechanism implemented in one of the |ddsrecorder| base libraries.
It is also possible to configure the number of threads that execute these data reception and saving tasks, as well as the size of the internal buffers to avoid writing to disk with each received data.

Usage Description
=================

The |ddsrecorder| is a terminal (non-graphical) application that creates a recording service as long as it is running.
Although most use cases are covered by the default configuration, the |ddsrecorder| can be configured via a YAML file,
whose format is very intuitive and human-readable.

* **Run**:
  Only the command that launches the application (``ddsrecorder``) needs to be executed to run a |ddsrecorder|.
  Please, read this :ref:`section <usage_configuration>` to apply a specific configuration, and this :ref:`section <usage_usage_application_arguments>` to see the supported arguments.
* **Interact**:
  Once the |ddsrecorder| application is running, the allowlist and blocklist topic lists could be changed in runtime by just changing the YAML configuration file.

>   It is also possible to change the status of the DDS Recorder (running, paused or stopped) by remote control of the application.

Reviewer: ```suggestion
  It is also possible to change the status of the recorder (running, paused or stopped) by remote control of the application.
```

  This remote control is done by sending commands via DDS or by using the graphical remote control application provided with the |eddsrecord| sofware tool (see :ref:`Remote control <remote_control>`).
* **Stop**:
  To stop the |ddsrecorder| application just send a `Ctrl+C` signal to stop de process gracefully (see :ref:`Close DDS Recorder <usage_stop_recording>`) or close it remotely using the remote control application (see :ref:`Remote control <remote_control>`).

Common Use cases
================

> To get started with the DDS Recorder, please visit section :ref:`getting_started_usage_example`.

Reviewer: ```suggestion
To get started with the |eddsrecord|, please visit section :ref:`getting_started_usage_example`.
```

Author: |ddsrecorder|

> In addition, this documentation provides several tutorials on how to set up the |ddsrecorder|, a comprehensive Fast DDS application using DynamicTypes and how to read the generated MCAP file.

Reviewer: ```suggestion
In addition, this documentation provides several tutorials on how to set up a |ddsrecorder|, a comprehensive Fast DDS application using DynamicTypes and how to read the generated MCAP file.
```
